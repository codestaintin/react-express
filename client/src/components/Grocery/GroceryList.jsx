import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import GroceryListRow from './GroceryListRow';

const GroceryList = ({ groceries, handleDelete, handlePurchase }) => (
  <Fragment>
    <div className="container">
      <table className="table table-hover main-section">
        <thead>
          <tr>
            <th scope="col">&nbsp;</th>
            <th scope="col">Grocery Name</th>
            <th scope="col">Purchase Status</th>
            <th scope="col">Action</th>
          </tr>
        </thead>
        <tbody>
          {
            groceries.map(grocery => (
              <GroceryListRow
                key={grocery._id}
                grocery={grocery}
                handleDelete={handleDelete}
                handlePurchase={handlePurchase}
              />
            ))
          }
        </tbody>
      </table>
    </div>
  </Fragment>
);

GroceryList.propTypes = {
  groceries: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handlePurchase: PropTypes.func.isRequired
};

export default GroceryList;
