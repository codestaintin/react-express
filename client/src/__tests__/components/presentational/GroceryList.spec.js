import React from 'react';
import { shallow } from 'enzyme';
import GroceryList from '../../../components/Grocery/GroceryList';

describe('<GroceryList>', () => {
  const props = {
    groceries: [],
    handleDelete: jest.fn(() => {}),
    handlePurchase: jest.fn(() => {})
  };

  it('renders the GroceryList', () => {
    const tree = shallow(<GroceryList {...props} />);
    expect(tree).toMatchSnapshot();
  });
});