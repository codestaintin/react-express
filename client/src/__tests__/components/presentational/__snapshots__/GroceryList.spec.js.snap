// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroceryList> renders the GroceryList 1`] = `
<Fragment>
  <div
    className="container"
  >
    <table
      className="table table-hover main-section"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Â 
          </th>
          <th
            scope="col"
          >
            Grocery Name
          </th>
          <th
            scope="col"
          >
            Purchase Status
          </th>
          <th
            scope="col"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</Fragment>
`;
